# Competition configuration options
options:
  insert_models_over_bins: true
  insert_models_over_stations: true
  briefcase_over_stations: true
  belt_population_cycles: 1

agv_infos:
  agv1:
    location: AS3
    products:
      part_0:
        type: assembly_battery_blue
        pose:
          xyz: [0.2, -0.15, 0]
          rpy: [0, 0, '-pi/4']
  agv2:
    location: KS2
    # products:
    #   part_0:
    #     type: assembly_battery_red
    #     pose:
    #       xyz: [0.2, -0.15, 0]
    #       rpy: [0, 0, 'pi/4']
  agv3:
    location: AS4
    # products:
    #   part_0:
    #     type: assembly_pump_blue
    #     pose:
    #       xyz: [0.2, -0.15, 0]
    #       rpy: [0, 0, 'pi/4']
  agv4:
    location: KS4
    
orders:
  order_0:
    task: kitting
    announcement_condition: time
    announcement_condition_value: 0.0
    shipment_count: 1
    destinations: [agv1]
    products:
      part_0:
        type: assembly_battery_blue
        pose:
          xyz: [0.1, -0.1, 0]
          rpy: [0, 0, 0]
  # Example for tasking one robot to do assembly and one robot to do kitting
  # order_1:
  #   task: kitting
  #     announcement_condition: time
  #     announcement_condition_value: 0.0
  #     shipment_count: 1
  #     destinations: [agv1]
  #     products:
  #       part_0:
  #         type: assembly_battery_blue
  #         pose:
  #           xyz: [0.1, -0.1, 0]
  #           rpy: [0, 0, 0]
  #   task: assembly
  #     announcement_condition: time
  #     announcement_condition_value: 0.0
  #     task_count: 1
  #     destinations: [station4]
  #     products:
  #       part_0:
  #         type: assembly_battery_blue
  #         pose:
  #           xyz: [-0.032465, 0.174845, 1.35]
  #           rpy: [0, 0, 0]
          
models_over_bins:
  #  bin1:
  #    models:
  #      assembly_regulator_red:
  #        xyz_start: [0.2, 0.2, 0.0]
  #        xyz_end: [0.4, 0.4, 0.0]
  #        rpy: [0, 0, 0]
  #        num_models_x: 2
  #        num_models_y: 2
   bin2:
     models:
       assembly_battery_blue:
         xyz_start: [0.2, 0.2, 0.0]
         xyz_end: [0.4, 0.4, 0.0]
         rpy: [0, 0, 0]
         num_models_x: 2
         num_models_y: 2
  #  bin3:
  #    models:
  #      assembly_pump_green:
  #        xyz_start: [0.2, 0.2, 0.0]
  #        xyz_end: [0.4, 0.4, 0.0]
  #        rpy: [0, 0, 0]
  #        num_models_x: 2
  #        num_models_y: 2
  #  bin4:
  #    models:
  #      assembly_battery_red:
  #        xyz_start: [0.2, 0.2, 0.0]
  #        xyz_end: [0.4, 0.4, 0.0]
  #        rpy: [0, 0, 'pi/4']
  #        num_models_x: 2
  #        num_models_y: 1

models_over_stations:
  station1:
    models:
      assembly_battery_green:
        xyz: [-0.032465, 0.174845, 0.15]
        rpy: [0, 0, 0]
      assembly_pump_red:
        xyz: [0.032085, -0.152835, 0.15]
        rpy: [0, 0, 0]
  station2:
    models:
      assembly_battery_red:
        xyz: [-0.032465, 0.174845, 0.15]
        rpy: [0, 0, 0]
  station3:
    models:
      assembly_pump_blue:
        xyz: [0.032085, -0.152835, 0.25]
        rpy: [0, 0, 0]
 
belt_models:
  piston_rod_part_blue:
    1.0:
      pose:
        xyz: [0.0, 4.3, 0.92]
        rpy: [0, 0, 'pi/2']
  gear_part_red:
    2.0:
      pose:
        xyz: [0.0, 4.3, 0.92]
        rpy: [0, 0, 'pi/2']

time_limit: -1
